### PR TITLE
Marking static variables final that are never modified

### PR DESCRIPTION
**Pull request by upthewaterspout**
To help reduce the number of mutable statics to look through, adding
final to all of the statics that are not actually mutated outside of a
static initializer.

This change *only* consists of adding final to these variables.

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by upthewaterspout** [comment]
@Bill

**Comment by pivotal-jbarrett** [comment]
@upthewaterspout The commit needs a JIRA

**Comment by galen-pivotal** [comment]
If tests and build pass, I don't know what can go wrong! (apart from reflection... but I think the JVM will guarantee that it won't fail even if it un-JITs some stuff?)

File: geode-common/src/main/java/org/apache/geode/cache/ExpirationAction.java [hunk 1]
```diff
@@ -164,7 +164,7 @@ public static ExpirationAction fromXmlString(String xmlValue) {
 
   // The 4 declarations below are necessary for serialization
   private static int nextOrdinal = 0;
-  public final int ordinal = nextOrdinal++;
+  public int ordinal = nextOrdinal++;
```
Comment (kirklund) [inline geode-common/src/main/java/org/apache/geode/cache/ExpirationAction.java]: Why remove final from `public int ordinal` if there's no code that reassigns it?

If this is the fix for a serialization bug, then I suggest filing a separate ticket and PR for that bug with a test for it.

File: geode-core/src/main/java/org/apache/geode/cache/client/internal/ServerDenyList.java [hunk 2]
```diff
@@ -56,7 +56,7 @@
   protected final ListenerBroadcaster broadcaster = new ListenerBroadcaster();
 
   // not final for tests.
-  static int THRESHOLD = Integer
+  static final int THRESHOLD = Integer
```
Comment (kirklund) [inline geode-core/src/main/java/org/apache/geode/cache/client/internal/ServerDenyList.java]: This constant is never used outside the class and can also be changed to `private`

File: extensions/geode-modules-session-internal/src/main/java/org/apache/geode/modules/session/internal/filter/Constants.java [hunk 3]
```diff
@@ -20,9 +20,9 @@
  */
 public class Constants {
 
-  public static String GEMFIRE_SESSION_REQUEST = "_gemfire_session_request_";
+  public static final String GEMFIRE_SESSION_REQUEST = "_gemfire_session_request_";
```
Comment (kirklund) [inline extensions/geode-modules-session-internal/src/main/java/org/apache/geode/modules/session/internal/filter/Constants.java]: GEMFIRE_SESSION_REQUEST is unused and should be deleted.

File: extensions/geode-modules-session-internal/src/main/java/org/apache/geode/modules/session/internal/filter/Constants.java [hunk 4]
```diff
@@ -20,9 +20,9 @@
...
 
-  public static String SESSION_STATISTICS_MBEAN_NAME =
+  public static final String SESSION_STATISTICS_MBEAN_NAME =
```
Comment (kirklund) [inline extensions/geode-modules-session-internal/src/main/java/org/apache/geode/modules/session/internal/filter/Constants.java]: SESSION_STATISTICS_MBEAN_NAME should be changed to package-private.

File: extensions/geode-modules-session/src/main/java/org/apache/geode/modules/session/filter/SessionCachingFilter.java [hunk 5]
```diff
@@ -74,17 +74,17 @@
   /**
    * Can be overridden during testing.
    */
-  private static AtomicInteger started = new AtomicInteger(
+  private static final AtomicInteger started = new AtomicInteger(
       Integer.getInteger(DistributionConfig.GEMFIRE_PREFIX + "override.session.manager.count", 1));
 
-  private static int percentInactiveTimeTriggerRebuild = Integer
+  private static final int percentInactiveTimeTriggerRebuild = Integer
```
Comment (kirklund) [inline extensions/geode-modules-session/src/main/java/org/apache/geode/modules/session/filter/SessionCachingFilter.java]: percentInactiveTimeTriggerRebuild is unused and should be deleted.

File: geode-core/src/main/java/org/apache/geode/internal/cache/TombstoneService.java [hunk 6]
```diff
@@ -322,10 +323,10 @@ public Object getBlockGCLock() {
...
   private static class Tombstone extends CompactVersionHolder {
     // tombstone overhead size
-    public static int PER_TOMBSTONE_OVERHEAD = ReflectionSingleObjectSizer.REFERENCE_SIZE // queue's
-                                                                                          // reference
-                                                                                          // to the
-                                                                                          // tombstone
+    public static final int PER_TOMBSTONE_OVERHEAD = ReflectionSingleObjectSizer.REFERENCE_SIZE // queue's
+        // reference
+        // to the
+        // tombstone
```
Comment (Bill) [inline geode-core/src/main/java/org/apache/geode/internal/cache/TombstoneService.java]: recommend reformatting as a multi-line comment

File: geode-common/src/main/java/org/apache/geode/cache/ExpirationAction.java (hunk 1 above)
Comment (upthewaterspout) [reply geode-common/src/main/java/org/apache/geode/cache/ExpirationAction.java]: Good catch! I must have accidentally removed final from this ordinal while I was fixing compilation errors related to adding other final clauses (I tried to make nextOrdinal final). I double checked the diff, this is the only place I removed final from something that was. I'll fix it.

**Pull request merged**